src/reference/02-DetailTopics/05-Plugins-and-Best-Practices/02-Plugins.md

```diff
@@ -508,6 +508,29 @@ libraryDependencies ++=
         Seq.empty
 ```
 
+### Publishing a plugin
+
+Sbt plugins can be published as any other projects. 
+
+However, there is one caveat if you attempt to publish your plugin to a repository that follows the Maven scheme.
+For many years, Sbt plugins have utilized an artifact layout that is incompatible with Maven-based repositories, and rectifying this in a fully compatible manner has proven to be quite challenging.
```

Comment:
```suggestion
For many years, sbt plugins have utilized an artifact layout that is incompatible with Maven-based repositories, and rectifying this in a fully compatible manner has proven to be quite challenging.
```

Reply:
I feel like this sentence is simultaneously too much detail for someone who might be interested in creating a new plugin, and not enough details to accurate capture the problem with pre-1.9 publishing. I don't know if it's clear what was "incompatible" with what, for instance. Maven Central has allowed sbt to publish, and sbt versions going back for many years sbt using either Ivy or Coursier was able to resolve plugins out of Maven Central, so it was compatible in that sense.

The problem with pre-1.9 sbt was that it published in a compatible, but intentionally invalid URL such as https://repo1.maven.org/maven2/com/eed3si9n/sbt-assembly_2.12_1.0/2.1.4/sbt-assembly-2.1.4.pom because the `artifactId` of the module is called sbt-assembly:

```xml
<artifactId>sbt-assembly</artifactId>
```

This was a workaround to preserve the `name` from sbt as `artifactId`, while allowing Apache Ivy to find the `_2.12_1.0` variant of sbt-assembly from a Maven repository. In an Ivy-layout repository, this concept is expressed as extra attributes, and therefore sbt team has initially recommended hosting sbt plugins on Ivy repository on Artifactory or Bintray. Preserving the name of module allows sbt to round-trip the POM back to our idea of modules, but also it has allowed external services such as https://search.maven.org/search?q=sbt-assembly to treat sbt-assembly as "sbt-assembly", not "sbt-assembly_2.12_1.0" etc.

This invalidity have been ok, apart from the fact that Artifactory has a checkbox called "Suppress POM Consistency Checks", and tried to enforce consistency of the URL layout. Post-1.9 sbt will double-publish to https://repo1.maven.org/maven2/com/eed3si9n/sbt-assembly_2.12_1.0/2.1.4/sbt-assembly_2.12_1.0-2.1.4.pom so the URL as well as the `artifactId` are now valid:

```xml
<artifactId>sbt-assembly_2.12_1.0</artifactId>
```

Reply:
I decided to drop this sentence as it indeed wasn't bringing much value.